lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -865,6 +865,38 @@ def upload_students_csv(_xmodule_instance_args, _entry_id, course_id, task_input
     return task_progress.update_task_state(extra_meta=current_step)
 
 
+def upload_may_enroll_csv(_xmodule_instance_args, _entry_id, course_id, task_input, action_name):
+    """
+    For a given `course_id`, generate a CSV file containing
+    information about students who may enroll but have not done so
+    yet, and store using a `ReportStore`.
+    """
+    start_time = time()
+    start_date = datetime.now(UTC)
+    num_students = CourseEnrollmentAllowed.num_may_enroll_and_unenrolled(course_id)
```

Comment:
I'm not certain this makes sense. This does the calculation we want in advance of doing the calculation we want, all to obtain a number for the task progress.... sorry now I'm dizzy :smile: 

But yeah - I'm not certain this is the best use of resources.